tests/M1_Checkout.php

```diff
@@ -0,0 +1,837 @@
+            ],
+        ];
+    }
+
+    /**
+     * Set customer info
+     *
+     * @param bool $isNewCustomer
+     * @throws Exception
+     */
+    protected function setCustomer($isNewCustomer = true)
+    {
+        $email = uniqid('test', true) . '@example.com';
+
+        if (!$isNewCustomer) {
+            $customer   = $this->getTestConfig()->getValue('customer');
+            $email      = !empty($customer['login']) ? $customer['login'] : $email;
+        }
+
+        $this->byId('firstname')->value('TestName');
+        $this->byId('lastname')->value('TestLastName');
+        $this->byId('account_email')->value($email);
+    }
+
+    /**
+     * Set payment info with valid or not card number
+     *
+     * @param bool $isBadCard
+     */
+    protected function setPayment($isBadCard = false)
+    {
+        $this->byId('cryozonic_stripe_cc_owner')->value('Test Card');
+        $this->byId('cryozonic_stripe_cc_number')->value($isBadCard ? '4000000000000002' : '4242424242424242');
+        $this->select($this->byId('cryozonic_stripe_expiration'))
+            ->selectOptionByValue(1);
+        $this->select($this->byId('cryozonic_stripe_expiration_yr'))
+            ->selectOptionByValue(2020);
+        $this->byId('cryozonic_stripe_cc_cid')->value('555');
+    }
+
+    /**
+     * Set shipping address
+     * With wrong or not postal code
+     * Return address for check
+     *
+     * @param bool $isPostalBad
+     * @return array
+     */
+    protected function setShippingAddress($isPostalBad = true)
+    {
+        $this->byId('shipping:street1')->value('1250 N Us Highway 1');
+
+        $this->select($this->byId('shipping:region_id'))
+            ->selectOptionByLabel('Florida');
+
+        $this->byId('shipping:city')->value('Pompano Beach');
+        $this->byId('shipping:postcode')->value($isPostalBad ? '33061' : '33062-3705');
+
+        $this->byId('shipping:telephone')->value('5555555555');
+
+        $address['street'][]    = $this->byId('shipping:street1')->value();
+        $address['region_id']   = $this->byId('shipping:region_id')->value();
+        $address['city']        = $this->byId('shipping:city')->value();
+        $address['postcode']    = $this->byId('shipping:postcode')->value();
+
+        return $address;
+    }
+
+    /**
+     * Set billing address
+     * Select address from Auto-complete Address Suggestions
+     * With wrong or not postal code
+     * Set shipping same as billing
+     * Return address for check
+     *
+     * @param bool $isPostalBad
+     * @param bool $isSame
+     * @return array
+     */
+    protected function setBillingAddress($isPostalBad = true, $isSame = true)
+    {
+        // Set address
+        $this->byId('billing:street1')->value('Houston Street');
+
+        sleep(3);
+
+        // Use keys for select
+        $this->keys(PHPUnit_Extensions_Selenium2TestCase_Keys::DOWN);
+        $this->keys(PHPUnit_Extensions_Selenium2TestCase_Keys::ENTER);
```

Comment:
The billing and the shipping forms both have Google Place Suggestions. Seems that we're testing the address suggestions on billing, but not on shipping. We should consider testing the google address suggestions on both forms. We should also consider what we might gain by testing non-auto suggest addresses in both forms.